eox_tagging/models.py

```diff
@@ -47,35 +49,44 @@ def invalid(self):
         """Returns all invalid tags."""
         return self.exclude(invalidated_at=None)
 
-    def find_by_owner(self, owner):
+    def find_by_owner(self, owner_type, owner_id):
         """Returns all valid tags owned by owner_id."""
-        model = owner.__class__
-        ctype = ContentType.objects.get_for_model(model)
 
-        return self.valid().filter(
+        owner, ctype = self.__get_object_for_this_type(owner_type, owner_id)
+
+        return self.filter(
             owner_type=ctype,
             owner_object_id=owner.id,
         )
 
-    def find_all_tags_for(self, target_object):
-        """Returns all valid on an object."""
-        if get_model_name(target_object) in OPAQUE_KEY_PROXY_MODEL_TARGETS:
-            model = OpaqueKeyProxyModel
-            target_id = OpaqueKeyProxyModel.objects.get(opaque_key=target_object.course_id).id
-        else:
-            model = target_object.__class__
-            target_id = target_object.id
-        ctype = ContentType.objects.get_for_model(model)
+    def find_all_tags_for(self, target_type, target_id):
```

Comment:
what is the  difference between this method and `find_by_owner`, only the `if` ? if both methods make the same, why not create a unique method called find_tag(self, type, filter) ???

Reply:
In find_by_owner I filter using owner_type and owner_object_id and in find_all_tags_for I filter using target_type and target_object_id. Do you think I should change it?

Reply:
It can be something like this? @andrey-canon 

```
def find_tags(self, object_type, object_id, role):

        object_type = PROXY_MODEL_NAME if object_type in OPAQUE_KEY_PROXY_MODEL_TARGETS else object_type
        instance_object, ctype = self.__get_object_for_this_type(object_type, object_id)

        if role is "target":
            data_for_filter = {
                "target_type": ctype,
                "target_object_id": instance_object.id,
            }
        else:
            data_for_filter = {
                "owner_type": ctype,
                "owner_object_id": instance_object.id,
            }

        return self.filter(
            **data_for_filter
        )
```

Reply:
I didn't notice that the return statement is different, so please forget my comment  and  keep the current version

Reply:
Okok. Thank you :)